File: CSharp/Tests/VowelReplacerTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class VowelReplacerTest
    {
        [Theory]
        [InlineData("the aardvark", "#", "th# ##rdv#rk")]
        [InlineData("minnie mouse", "?", "m?nn?? m??s?")]
        [InlineData("shakespeare", "*", "sh*k*sp**r*")]
        public void ReplaceVowels_StringInputValueStringReplaceValue_ReturnStringWithVowelsReplacedWithInputCharacter(string str, string ch, string expected)
        {
            var actual = VowelReplacer.ReplaceVowels(str, ch);

            Assert.Equal(expected, actual);
        }
    }
}

File: js/tests/vowelReplacer.test.js
const replaceVowels = require("../vowelReplacer");

const data = [
  ["the aardletk", "#", "th# ##rdv#rk"],
  ["minnie mouse", "?", "m?nn?? m??s?"],
  ["shakespeare", "*", "sh*k*sp**r*"],
  ["all is fair in love and war", "<", "<ll <s f<<r <n l<v< <nd w<r"]
];

describe("replaceVowels", () => {
  describe("Replace all vowels in string with a specified character", () => {
    test.each(data)("replaceVowels(%p, %p) = %p", (str, ch, result) => {
      let actual = replaceVowels(str, ch);

      expect(actual).toEqual(result);
    });
  });
});
